docs/evaluation_run_lifecycle.md

```diff
@@ -0,0 +1,46 @@
+# Evaluation Run Lifecycle
+
+Each invocation of `helm-run` will run a number of **evaluation runs**. Each evaluation run uses a single scenario and a single model, and is performed independently form other evaluation runs. Evaluation runs are usually executed serially / one at a time by the default runner, though some alternate runners (e.g. `SlurmRunner`) may execute evaluation runs in parallel.
+
+## Evaluation steps
+
+An evaluation run has the following steps:
+
+1. Get in-context learning and evaluation instances from scenario. Each instance has an input (e.g. question) and a set of reference outputs (e.g. multiple choice options).
+2. (Advanced) Run _data augmenters / perturbations_ on the base instances to generate perturbed instances.
+3. Perform _adaptation_ to transform the in-context learning instances and evaluation instances into model inference requests, which contain prompts and other request parameters such as request temperature and stop sequences.
+4. Send the requests to the models and receives the request responses.
+5. Compute the per-instance stats and aggregate them to per-run stats.
+
+The following code and data objects are responsible involved in an evaluation run:
```

Comment:
`responsible involved in ` should either be `responsible for` or just `involved in`